docs/semgrep-secrets/conceptual-overview.md

```diff
@@ -0,0 +1,111 @@
+
+Semantic analysis refers to Semgrep's ability to understand how data is used within your code. This differentiates Semgrep from regex-based detectors that simply define a pattern to match to a piece of code.
+
+Semgrep uses several mechanisms to perform semantic analysis. In particular, Semgrep uses **data-flow analysis**, which means that it is able to track data, such as variables, and the flow of that data across files and functions in your codebase. Semgrep is able to detect if a variable is renamed, unsanitized or sanitized, reassigned, or used in a function in such a way that a secret is exposed.
+
+See the following Semgrep rule and JavaScript test code for an example.
+
+<iframe title="AWS hardcoded access key" src="https://semgrep.dev/embed/editor?snippet=EPj5" width="100%" height="432px" frameBorder="0"></iframe>
+<br />
+
+ The Semgrep rule detects hardcoded AWS secret access keys. The test code defines an access key in the variable `secret`. Click **<i class="fa-solid fa-play"></i> Run** to see the true positives.
+
+* **Line 2:** Semgrep and regex-based scanners can detect both the variable name `secret` and its value (token) in line 2.
+	* A regex-based scanner may generate a noisy finding from line 2 even though `secret` has not been passed to any function.
+	* Semgrep doesn't generate a finding because the token hasn't been passed as a `secretAccessKey` or similar.
+* **Line 7:** Semgrep and regex-based scanners are able to detect **line 6**, in which the plain-text secret is passed to the `AWS.config.update` function.
+* **Line 17:** Semgrep and regex-based scanners can detect **line 14**, in which `secret` is passed.
+* **Line 26:** Semgrep correctly skips `conf.secret` in **line 21**. Regex-based scanners simply looking for matches of the string `secret` generate a false positive.
+
+After scanning for secrets, Semgrep uses a **post-processor** function called a **validator** to validate a secret.
+
+1. The post-processor detects the service, such as Slack or AWS, that the secret is used for.
+2. If the post-processor does not support the service that the secret is used for, or the secret is a **honeypot token**, Semgrep notes that there is **No validator** for the secret.
+3. If the validator can detect the service, Semgrep performs an API call. The following outcomes can occur:
+	1. If the API call returns an HTTP response of 200 or similar, then the finding's validation state is **confirmed valid**.
+	2. If the API call returns an HTTP response of 403 or similar, then the finding's validation state is **confirmed invalid**.
+	3. If the API call returns an HTTP response of 500 or similar, or some other error occurred, then Semgrep notes that a **validation error** has occurred.
+
+All findings, whether they are validated, with validation errors, invalid, or valid, appear in Semgrep Cloud Platform.
+
+By performing this post-processor check, you are able to prioritize (triage) the most high priority, active findings.
```

Comment:
We may want to add (next week) an example of what a validation rule looks like and highlight that people can create their own validation checks